src/bot_telegram.ts

```diff
@@ -160,13 +161,46 @@ export default class TelegramBot extends BotClient {
         this.logError(`Failed to send message to channel:\n${error}`);
       }
     } else {
-      let text = this.msgFromMarkdown(message.toMDString());
+      const link = message.title.link;
+      let text = '';
+      let templateFound = false;
+
+      const templates = message.game.telegramIVTemplates;
+      this.logDebug(`There are ${templates.length} templates for ${message.game.label}.`);
+
+      // Test for IV template matches
+      for (const telegramIVtemplate of templates) {
+        const templateLink = telegramIVtemplate.testUrl(link);
+        if (templateLink) {
+          this.logDebug(`IV template found for ${link}.`);
+          templateFound = true;
+          const titleText = `[${message.title.text}](${templateLink})`;
+
+          if (message.author.text) {
+            const authorText = message.author.link
+              ? `[${message.author.text}](${message.author.link})`
+              : message.author.text;
+
+            text = `New **${message.game.label}** update - ${authorText}:\n\n${titleText}`;
+          } else {
+            text = `New **${message.game.label}** update:\n\n${titleText}`;
+          }
+        }
+      }
+
+      // Check if an IV template matched
+      if (!templateFound) {
+        this.logDebug(`IV template NOT found for ${link}.`);
+        // Convert to normal text
+        text = this.msgFromMarkdown(message.toMDString());
+      }
+
       if (text.length > 2048) {
         text = text.substring(0, 2048);
       }
       try {
         await this.bot.sendMessage(channel.id, text, {
-          disable_web_page_preview: true,
+          disable_web_page_preview: !templateFound,
```

Comment:
So if there's no template we still keep the preview? I think I prefer no preview at all even if there's no template. For example for reddit posts it will usually be empty or wip anyway

Reply:
We use the preview if there is a template (because otherwise the IV won't show and we cut off the noticiation content anyway) and it's disabled if we didn't find a template because of bad previews and duplicate information.

Reply:
I completely misunderstood this, ignore